### ISSUE TITLE
Feature: New round button.

### ISSUE THREAD

**Issue by Davst** [johreh/gloomycompanion#64]
A new round button, that draws next card for all monsters and reshuffles any cards with shuffle at end of round would be very nice.

**Comment by stellentus** [comment]
I agree this would be nice, but in most rooms, not all monsters will be active, so designing a good interface for it would be complicated.

**Comment by GinoGalotti** [comment]
I made this my own a while back, the only problem is that we require to "disable" decks in order to implement this. I asked on the FAQ thread and, for instance:

You're fighting a boss that summons Living Bones.
Round 1.
You kill all the living bones.
Round 2.
A new round happens, so you should draw a card for everyone BUT the living bones.
Round 3.
A new round happens, so you should draw a card for everyone BUT the living bones.
Round 4.
Boss summons more living bones. They should have the deck in the same state as Round 1. 
Round 5.
A new round happens, so you should draw a card for everyone.

That's why I discarded the new Round button.

**Comment by stellentus** [comment]
I've been thinking about this more, and I think it could be nice with the following changes:
1. Along the top of the screen is a list of toggle buttons for the cards in the scenario. This is how you enable/disable a deck.
2. There's a new button: "Begin Round". It automatically shuffles any decks that must be shuffled (including the attack deck) and flips cards for active monsters.
3. Possibly remove the shuffle button for the attack deck; it's no longer necessary.
4. Clicking on a deck brings up an "Are you sure?" dialog instead of immediately flipping the next card. This helps to prevent accidental clicks. (But it's not good to completely disable flipping individual decks. You need to be able to manually flip if you realize halfway through the round that there is actually a Living Bones on the board and you didn't enable its deck before starting the round.)
5. This would also allow for a space optimization (which is good for phone/tablet screens): since there's now a way of indicating a deck is active or inactive, hide the decks for monsters that are currently inactive (though state should be remembered in case those monsters show up again).

Since this automates a lot more, Undo (#65) should probably be implemented first.

**Comment by GinoGalotti** [comment]
It makes sense, but that requires a big change in how things work right now. That was my first idea with "shuffle button" when I implemented it until I found out the scenario I described before and our need to enable/disable decks. I always wanted a "one button new round" feature. These weeks are crazy, but maybe I'll try to invest some time on doing this.
